chain/epoch-manager/src/shard_tracker.rs

```diff
@@ -60,9 +73,18 @@ impl ShardTracker {
         match &self.tracked_shards_config {
             TrackedShardsConfig::NoShards => Ok(false),
             TrackedShardsConfig::AllShards => Ok(true),
+            TrackedShardsConfig::Shards(tracked_shards) => {
+                // TODO(#13445): Turn the check below into a debug assert and call it earlier,
+                // for all `tracked_shards_config` variants.
+                let shard_layout = self.epoch_manager.get_shard_layout(epoch_id)?;
+                if !shard_layout.shard_ids().contains(&shard_id) {
+                    return Err(EpochError::ShardingError(format!("Invalid shard id {shard_id}")));
+                }
+                self.check_if_descendant_of_tracked_shard(shard_id, tracked_shards, epoch_id)
+            }
             TrackedShardsConfig::Accounts(tracked_accounts) => {
                 let shard_layout = self.epoch_manager.get_shard_layout(epoch_id)?;
-                let tracking_mask = self.tracking_shards_cache.get_or_try_put(
+                let tracking_mask = self.tracked_accounts_shard_cache.get_or_try_put(
```

Comment:
nit: maybe also move this to a helper method like you've done with `check_if_descendant_of_tracked_shard`